experimental/FTheoryTools/src/AbstractFTheoryModels/attributes.jl

```diff
@@ -1195,33 +1186,222 @@ function chern_class_c2(m::AbstractFTheoryModel; check::Bool = true)
-    cl = toric_divisor_class(ambient_space(m), degree(weierstrass_polynomial(m)))
+  cy = cohomology_class(toric_divisor_class(ambient_space(m), degree(hypersurface_equation(m))))
+
+  # Identify index combination that may be discarded by virtue of the Stanley-Reisner ideal
+  mnf = Oscar._minimal_nonfaces(ambient_space(m))
+  indices = [findall(x -> x == 1, Vector{Int}(mnf[i, :])) for i in 1:Polymake.nrows(mnf)]
+
+  # Compute c2 of the toric ambient space
+  c2_t_ambient = zero(R)
+  for i in 1:len_c_ds - 1
+    for j in i+1:len_c_ds
+      [i,j] in indices && continue
+      c2_t_ambient = c2_t_ambient + c_ds[i] * c_ds[j]
+    end
   end
-  if m isa GlobalTateModel
-    cl = toric_divisor_class(ambient_space(m), degree(tate_polynomial(m)))
+  c2_t_ambient = cohomology_class(ambient_space(m), projection_map(c2_t_ambient))
+
+  # Compute and set h
+  h = c2_t_ambient - cy * chern_class_c1(m; check = check)
+  set_attribute!(m, :chern_class_c2, h)
+  return h
+end
+
+
+
+@doc raw"""
+    chern_class_c3(m::AbstractFTheoryModel; check::Bool = true)
+
+If the elliptically fibered n-fold $Y_n$ underlying the F-theory model in question is given
+as a hypersurface in a toric ambient space, we can compute a cohomology class $h$ on the
+toric ambient space $X_\Sigma$, such that its restriction to $Y_n$ is the 3rd Chern class
+$c_3$ of the tangent bundle of $Y_n$. If those assumptions are satisfied, this method returns
+this very cohomology class $h$, otherwise it raises and error.
+
+As of right now, this method is computationally quite expensive for involved toric varieties,
+such as in the example below. Therefore, think carefully if you truly want to compute this quantity.
+
+The computation of the cohomology ring verifies if the toric variety is simplicial and
+complete. The check for it to be complete can be very time consuming. This can be switched
+off by setting the optional argument `check` to the value `false`, as in the example below.
+
+```jldoctest; setup = :(Oscar.LazyArtifacts.ensure_artifact_installed("QSMDB", Oscar.LazyArtifacts.find_artifacts_toml(Oscar.oscardir)))
+julia> qsm_model = literature_model(arxiv_id = "1903.00009", model_parameters = Dict("k" => 4))
+Hypersurface model over a concrete base
+
+julia> h = chern_class_c3(qsm_model; check = false);
+
+julia> is_trivial(h)
+false
+```
+"""
+function chern_class_c3(m::AbstractFTheoryModel; check::Bool = true)
+  @req (m isa WeierstrassModel || m isa GlobalTateModel || m isa HypersurfaceModel) "Third Chern class of F-theory model supported for Weierstrass, global Tate and hypersurface models only"
+  @req base_space(m) isa NormalToricVariety "Third Chern class of F-theory model currently supported only for toric base"
+  @req ambient_space(m) isa NormalToricVariety "Third Chern class of F-theory model currently supported only for toric ambient space"
+
+  # Check if the answer is known
+  if has_attribute(m, :chern_class_c3)
+    return get_attribute(m, :chern_class_c3)
```

Comment:
Maybe like this?
```suggestion
    return get_attribute(m, :chern_class_c3)::CohomologyClass
```

Reply:
I understand why you suggest to add `::CohomologyClass` (thank you for the pointer).

But why remove the if statement? Is the check for `has_attribute` not necessary?